lib/uwb/UwbPeer.cxx

```diff
@@ -32,6 +33,47 @@ UwbPeer::UwbPeer(UwbMacAddress address) :
     m_address(std::move(address))
 {}
 
+/**
+ * @brief Assuming the Arm definition of Qm.n formatting, the most significant bit is the sign, the next
+ * (m-1) bits are an integer, and the next n bits is the number to be multiplied by pow(2,n)
+ * The double equivalent will be the sum of those two results
+ * TODO double check this conversion, write tests for it
+ *
+ * @param q97 a number in Q9.7 format
+ * @return double
+ */
+double
+ConvertQ9_7FormatToIEEE(uint16_t q97)
+{
+    static const double pow2 = std::pow(2, -7);
+    static const uint16_t sign_mask =               0b1000'0000'0000'0000U;
+    static const uint16_t unsigned_integer_mask =   0b0111'1111'1000'0000U;
+    static const uint16_t fraction_mask = ~(sign_mask | unsigned_integer_mask);
+
+    bool sign = q97 & sign_mask;
+    int unsigned_integer_part = (q97 & unsigned_integer_mask) >> 7U;
+    int fraction_part = q97 & fraction_mask;
+
+    double unsigned_number = ((double)unsigned_integer_part) + (((double)fraction_part) * pow2);
+
+    return (sign ? -1 : 1) * unsigned_number;
+}
+
+UwbPeer::UwbPeer(const uwb::protocol::fira::UwbRangingMeasurement& data) :
+    m_address{ data.PeerMacAddress },
+    m_spatialProperties{
+        .Distance{ data.Distance }, // TODO is this also q97
+        .AngleAzimuth{ ConvertQ9_7FormatToIEEE(data.AoAAzimuth.Result) },
+        .AngleElevation{ ConvertQ9_7FormatToIEEE(data.AoAElevation.Result) },
+        .Elevation{ ConvertQ9_7FormatToIEEE(data.AoaDestinationElevation.Result) }, // TODO is this right?
+
+        .AngleAzimuthFom{ data.AoAAzimuth.FigureOfMerit },
+        .AngleElevationFom{ data.AoAElevation.FigureOfMerit },
+        .ElevationFom{ data.AoaDestinationElevation.FigureOfMerit }
+    }
```

Comment:
Might be worth adding a c'tor to `UwbPeerSpatialProperties` that takes a `uwb::protocol::fira::UwbRangingMeasurement` as an input argument.

Reply:
I think it makes sense to hold off on implementing the reverse conversion because we may want to put all of the conversion functions in the same place to avoid circular header dependencies